pallets/treasury-funding/src/tests.rs

```diff
@@ -0,0 +1,94 @@
+use crate::mock::*;
+use frame_support::assert_ok;
+
+#[test]
+fn test_fund_treasury_without_vesting() {
+	new_test_ext().execute_with(|| {
+		let vault_account = TreasuryFunding::account_id();
+		let treasury_account = Treasury::account_id();
+
+		// Step 1: Fund the vault account with an initial balance.
+		let initial_balance = 1_000;
+		let _ = Balances::force_set_balance(RuntimeOrigin::root(), vault_account, initial_balance);
+
+		// Step 2: Verify the initial balances of the vault and treasury accounts.
+		assert_eq!(Balances::free_balance(vault_account), initial_balance);
+		assert_eq!(Balances::free_balance(treasury_account), 0);
+
+		// Step 3: Call the fund_treasury function to transfer funds.
+		assert_ok!(TreasuryFunding::fund_treasury(RuntimeOrigin::signed(vault_account)));
+
+		// Step 4: Verify the final balances after the transfer.
+		assert_eq!(Balances::free_balance(vault_account), 0);
+		assert_eq!(Balances::free_balance(treasury_account), initial_balance);
+	});
+}
+
+#[test]
+fn test_fund_treasury_during_active_vesting() {
+	new_test_ext().execute_with(|| {
+		let vault_account = TreasuryFunding::account_id();
+		let treasury_account = Treasury::account_id();
+
+		// Step 1: Fund the vault account with an initial balance.
+		let initial_balance = 1_000;
+		let _ = Balances::force_set_balance(RuntimeOrigin::root(), vault_account, initial_balance);
+
+		// Step 2: Create a vesting schedule for the vault account.
+		Vesting::vested_transfer(
+			RuntimeOrigin::signed(vault_account),
+			vault_account,
+			pallet_vesting::VestingInfo::new(1_000, 1, 0),
+		)
+		.unwrap();
```

Comment:
Instead of unwrap the result, we can check it's ok
```suggestion
		assert_ok!(Vesting::vested_transfer(
			RuntimeOrigin::signed(vault_account),
			vault_account,
			pallet_vesting::VestingInfo::new(1_000, 1, 0),
		));
```